src/components/calcite-slider/calcite-slider.scss

```diff
@@ -43,108 +52,195 @@ $tick-height: 4px;
   margin-bottom: $thumb-size + $thumb-padding;
 }
 
-// focus styles
-.thumb {
-  @include focus-style-base();
-  &:focus {
-    @include focus-style-outset();
-  }
-}
-
 .thumb {
   position: absolute;
-  height: $thumb-size;
-  width: $thumb-size;
-  margin: -15px;
-  box-sizing: border-box;
   border: none;
   background: transparent;
   cursor: pointer;
   font-family: inherit;
   z-index: 2;
-}
+  outline: none;
+  padding: 0;
+  display: flex;
+  flex-direction: column;
+  align-items: center;
+  transform: translate(7px, -8px);
 
-.handle {
-  position: absolute;
-  top: 0;
-  left: 0;
-  height: $handle-size;
-  width: $handle-size;
-  margin: $thumb-padding;
-  box-sizing: border-box;
-  border-radius: 100%;
-  background-color: var(--calcite-ui-foreground-1);
-  border: 2px solid var(--calcite-ui-text-3);
-  transition: border 0.25s ease, background-color 0.25s ease,
-    box-shadow 0.25s ease;
-}
-
-.handle__label {
-  position: absolute;
-  left: 0;
-  bottom: $thumb-size;
-  width: $thumb-size;
-  height: 0.75em;
-  @include font-size(-3);
-  font-weight: 500;
-  line-height: 1;
-  color: var(--calcite-ui-text-3);
-  text-align: center;
-}
+  .handle__label {
+    transition: transform 150ms;
+    @include font-size(-4);
+    font-weight: 500;
+    line-height: 1;
+    color: var(--calcite-ui-text-2);
+    margin-bottom: 5px;
+    &.static,
+    &.transformed {
+      opacity: 0;
+      position: absolute;
+      top: 0;
+    }
+    &--minValue.hyphen::after {
+      content: "\2014";
+      display: inline-block;
+      width: 1em;
+    }
+    &--minValue.hyphen.max-offset::after {
+      margin-right: 1em;
+    }
+    &--value.hyphen::before {
+      content: "\2014";
+      display: inline-block;
+      width: 1em;
+    }
+    &--value.hyphen.min-offset::before {
+      margin-left: 1em;
+    }
+  }
+
+  .handle {
+    @include focus-style-base();
+    height: $handle-size;
+    width: $handle-size;
+    box-sizing: border-box;
+    border-radius: 100%;
+    background-color: var(--calcite-ui-foreground-1);
+    box-shadow: 0 0 0 2px var(--calcite-ui-text-3) inset;
+    transition: border 0.25s ease, background-color 0.25s ease,
+      box-shadow 0.25s ease;
+  }
+
+  .handle-extension {
+    width: 2px;
+    height: $thumb-padding;
+    background-color: var(--calcite-ui-text-3);
+  }
+
+  &:hover {
+    .handle {
+      box-shadow: 0 0 0 3px var(--calcite-ui-blue-1) inset;
+    }
+    .handle-extension {
+      background-color: var(--calcite-ui-blue-1);
+    }
+  }
 
-.thumb:hover .handle {
-  border-width: 3px;
-  border-color: var(--calcite-ui-blue-1);
-  @include shadow(1, "hover");
+  &:focus {
+    .handle {
+      @include focus-style-outset();
+      outline-offset: 6px;
+    }
+    .handle-extension {
+      background-color: var(--calcite-ui-blue-1);
+    }
+  }
+}
+.thumb--minValue {
+  transform: translate(-7px, -8px);
```

Comment:
what are all these magic pixel values in the translate y position?

Reply:
The `7px` is half the handle size, so I suppose I could do `$handle-size / 2` for that and the `-8px` is the correct amount of `y` distance needed to place the center of the handle circle at the end of of the track line.  Since the handle is 7x7 you would think that the 8 could just be 7, but it needed an extra pixel for some reason.

Reply:
I found out why it has to be `8px` in the y direction: `.track { height: 2px }`.  Because the track element is set to 2px height, if you bump that down to 1 and change the transform y to 7px you'll see the circle centered on the track line, so that's why it needs to be 1 pixel more to get it to center.